core/src/main/java/org/testcontainers/images/builder/ImageFromDockerfile.java

```diff
@@ -63,6 +64,7 @@
     private final Map<String, Transferable> transferables = new HashMap<>();
     private final Map<String, String> buildArgs = new HashMap<>();
     private Optional<String> dockerFilePath = Optional.empty();
+    private Optional<File> dockerfile = Optional.empty();
```

Comment:
can we have Path here please?

Reply:
@bsideup do you mean hold an `Optional<Path>` object in the field _and_ make the with-setter accept a `Path`, or just change the field type?

Reply:
I have `File` here because the underlying docker-java API takes a `File`, so I thought it would be best to not do the unnecessary conversion. Is there a particular reason we should be using `Path` over `File`?

Reply:
`Path` is a bit more forward thinking abstraction, easier to work with and I hope docker-java will adopt `Path` API soon too, so that we can avoid temporary files when it is possible.
I would prefer to have it in Testcontainers, because that's our public API, and to not block us from making changes to it later :)